### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by misheck12**
Changes made in this pull request :

-  Added RSpec as a gem to your project.

-  Create a Request spec file for all your controllers.

### CODE REVIEW

File: db/migrate/20220706163605_fix_comment_counter_name.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class FixCommentCounterName < ActiveRecord::Migration[7.0]
+  def change
+    rename_column :posts, :comment_counter, :comments_counter
+    rename_column :users, :post_counter, :posts_counter
+  end
+end
```
Comment (ShoiraTa) [inline db/migrate/20220706163605_fix_comment_counter_name.rb:6]: - [x] Please check the migration files, when I am running rails db:migrate I am getting errors. perhaps you deleted some migrations manually. To reproduce the error you have to drop the DB and run migrations again. 

For this reason I cannot run the tests, they are also failing. 

Screenshot 2022-07-19 at 13 52 59

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @misheck12,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220706163605_fix_comment_counter_name.rb (hunk 1 above)
Comment (misheck12) [reply db/migrate/20220706163605_fix_comment_counter_name.rb:6]: Hey @ShoiraTa thank you for this review , be assured this has been worked on.

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @misheck12, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**